### PR TITLE
Aligned calloc routine (host and device)

### PR DESCRIPTION
**Pull request by jrreap**
Adds test for the aligned variant of the calloc routine. Tests both on device and host.

### CODE REVIEW

**Comment by jrreap** [comment]
`/autofs/nccs-svm1_home1/jreap/sollve_vv/tests/5.1/allocate/test_target_aligned_calloc.c:27:63: error: called object type 'omp_allocator_handle_t' (aka 'enum omp_allocator_handle_t') is not a function or function pointer 

#pragma omp target map(tofrom: errors) uses_allocators(alloc(traits))`

Throws this error for device test, seems confused by the custom allocator traits

File: tests/5.1/allocate/test_aligned_calloc.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    omp_free(x, alloc);
+  }
+
+  omp_destroy_allocator(alloc);
+
+  return errors;
+}
+
+int main() {
+  OMPVV_TEST_OFFLOADING;
```
Comment (fel-cab) [inline tests/5.1/allocate/test_aligned_calloc.c:66]: This is a host test, so we should remove the `OMPVV_TEST_OFFLOADING`

File: tests/5.1/allocate/test_aligned_calloc.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+#include "ompvv.h"
+
+#define N 1024
+
+int test_aligned_calloc() {
+  int errors = 0;
+  
+  omp_memspace_handle_t  memspace = omp_default_mem_space;
+  omp_alloctrait_t       traits[1] = {{omp_atk_alignment, 64}};
+  omp_allocator_handle_t alloc = omp_init_allocator(memspace,1,traits);
```
[the remark this answers is not in the document]
Comment (spophale) [reply tests/5.1/allocate/test_aligned_calloc.c:25]: Why ?

**Comment by spophale** [comment]
We need to check if %64 is the correct way to go to check alignment requirement. Spec says:
```
Memory allocated by omp_aligned_calloc will be byte-aligned to at least the maximum of
the alignment required by malloc, the alignment trait of the allocator and the alignment argument value.
```

File: tests/5.1/allocate/test_aligned_calloc.c (hunk 1 above)
Comment (jrreap) [reply tests/5.1/allocate/test_aligned_calloc.c:66]: Good catch! I'll patch it up.

**Comment by jrreap** [comment]
The %64 actually comes from their `omp_alloc` example in the 5.1 examples document. They are passing special alignment parameters to the allocator and check if it's properly aligned in a similar way. It's in section 10.2 "Memory Allocators" (page 426 in the online PDF)

**Comment by jrreap** [comment]
Get nvlink errors with cuda 11.5.2 and llvm 17.0.0-latest on summit. Seems offloading may just not be supported yet?

`nvlink error   : Undefined reference to 'omp_aligned_calloc' in '/tmp/test_target_aligned_calloc-cadfa3-nvptx64-nvidia-cuda-sm_70-d4219b.cubin'`

**Comment by jrreap** [comment]
Both tests run as expected and pass on gcc 12.2.1 with cuda 11.5.1. Looks like clang just does not support the aligned on device quite yet. Should be good to go now.

**Review by spophale — APPROVED** [review]

**Review by fel-cab — APPROVED** [review]

**Pull request merged**